src/System.Threading.Thread/src/System/Threading/Thread.cs

```diff
@@ -21,6 +21,7 @@ public sealed partial class Thread : CriticalFinalizerObject
         private CultureInfo _startCulture;
         private CultureInfo _startUICulture;
         private IPrincipal _principal;
```

Comment:
(Not related to your change) This may look better as ` [ThreadStatic]` - saves fetching the current thread on access.